src/lib.rs

```diff
@@ -15,9 +15,14 @@ pub struct PopplerPage(*mut ffi::PopplerPage);
 impl PopplerDocument {
     pub fn new_from_file<P: AsRef<path::Path>>(
         p: P,
-        password: &str,
+        password: Option<&str>,
     ) -> Result<PopplerDocument, glib::error::Error> {
-        let pw = CString::new(password).map_err(|_| {
+        let pw = CString::new(if password.is_none() {
+            ""
+        } else {
+            password.expect("That it is something when check if it is.")
```

Comment:
Can you please improve the error message?

Reply:
Changed in commit version [6917c0d](https://github.com/DMSrs/poppler-rs/pull/13/commits/6917c0da6d83819b020bf1d7f731caa3d2c136ad)